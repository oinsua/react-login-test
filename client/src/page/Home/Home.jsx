import React from 'react';
import { useParams} from "react-router-dom";

const Home = () => {

    const {username} = useParams();

    return (
        <div>
            <h1>`User name is ${username}`</h1>
        </div>
    )
}

export default Home
