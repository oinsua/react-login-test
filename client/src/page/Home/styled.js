import styled from '@emotion/styled';
import {Link as LinkRouter} from "react-router-dom";
export const Link = styled(LinkRouter)`
                                    margin-top: 15px;
                                    padding: 10px;
                                    color: var(--brand-color_1);
                                    font-size: 18px;
                                    font-weight: bold;
                                    text-decoration: none;
                                    cursor: pointer;

                                    &:hover{
                                        color: var(--brand-color_4); 
                                    }
`