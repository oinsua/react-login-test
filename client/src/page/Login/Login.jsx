import React, {useState} from 'react';
import {Helmet} from 'react-helmet';
import {useHistory} from 'react-router-dom';
import {SingIn} from '../../service/getSingIn';
import Modal from '../../components/Modal/Modal';


//Importando los Styled Component
import {Form, DivInput, Input, Button, Span, SpanLink, SpanError} from './styled';
 
const Login = () => {
    
    const history = useHistory(); //Se crea el objeto para la navegacion
    const [user, setUser] = useState({username: '', password: ''}); //Inicializo el estado con user y password
    const [showModal, setShowModal] = useState(false);
    
    const handleClick = () => {
        return setShowModal(true);
      };
    
      const handleClose = () => {
        setShowModal(false);
      };
    
    const [error, setError] = useState('');
   const handleInputChange = (e) => {
          setUser({
              ...user,
              [e.target.name]: e.target.value
          })
   }

   const [jwt, setJwt] = useState('');

   const handleSubmit = (e) => {
       e.preventDefault();

       SingIn({username: user.username, password: user.password})
        .then(jwt => setJwt(jwt))
        .catch( error => 
            setError('User not exist or password incorrect!')
        )
       if(error !== ''){ //Validar que los campos no estan vacio
            history.push(`/home/${user.username}`);
            localStorage.setItem(user.username, jwt);        
    }
   }

    return (
        <>
            <Helmet>
              <title>Login_Web</title>
              <meta name="application-name" content="login_Web"/>
              <meta name="description" content="login_Web"/>
              <meta name="google" content="notranslate"/>
            </Helmet>
            <Form onSubmit={handleSubmit}>
                <DivInput>
                    <Input placeholder="Your username" name="username" type="text" onChange={handleInputChange} value={user.username}/>
                    <Input placeholder="Your Password " name="password" type="password" onChange={handleInputChange} value={user.password}/>
                    <Button name="login" type="submit">LOGIN</Button>
                </DivInput>
                <Span>You can Register at this Web Application</Span>
               <SpanLink onClick={handleClick}>Register</SpanLink>
                {showModal && (
                    <Modal onClose={handleClose}>
                    </Modal>
                  )}
              <SpanError>{error}</SpanError>
            </Form>
            
        </>
    )
}

export default Login
