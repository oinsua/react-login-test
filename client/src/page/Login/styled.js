import styled from '@emotion/styled';

export const Form = styled.form`
                             display: flex;
                             flex-direction: column;
                             justify-content: center; 
                             align-items: center;
                             background-color: var(--brand-color_4);
                             border: 3px solid var(--brand-color_15);
                             border-radius: 10px;
                             margin: 10px;
                             width: 95%;
                             height: 250px;
`

export const SpanLink = styled.span`
                                    margin-top: 15px;
                                    padding: 10px;
                                    color: var(-brand-color_12);
                                    font-size: 18px;
                                    text-decoration: none;
                                    
                                    &:hover{
                                        color: var(--brand-color_8); 
                                    }
`

export const DivInput = styled.div`
                                   display: flex;
                                   flex-direction: column;
                                   justify-content: center;
                                   align-items: center;
`

export const Input = styled.input`
                                   display: inline-block;
                                   width: 220px;
                                   margin: 5px 0px;
                                   padding: 5px;
                                   background-color: whitesmoke;
                                   border-radius: 10px;
                                   height: 28px;
`

export const Button = styled.button`
                                   display: inline-block;
                                   margin: 5px;
                                   width: 230px;
                                   margin: 5px 0px;
                                   padding: 5px;
                                   background-color: var(--brand-color_5);
                                   color: whitesmoke;
                                   letter-spacing: 3px;
                                   border-radius: 10px;
                                   height: 40px;
                                   border: 2px solid var(--brand-color_1);

                                   &:hover{
                                    background-color: var(--brand-color_6); 
                                   }
`

export const Span = styled.span`
                                color: whitesmoke;
                                font-weight: bold;
                                margin-top: 5px;
                                text-align: center'
`
