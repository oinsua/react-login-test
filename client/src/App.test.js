import { render, screen } from '@testing-library/react';
import {BrowserRouter} from 'react-router-dom'
import toJson from 'enzyme-to-json';
import Enzyme, { shallow} from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

import Login from './page/Login/Login';
import Home from './page/Home/Home';

Enzyme.configure({ adapter: new Adapter() });

test('render Login correctly', () => {
    const wrapper = shallow(
      <BrowserRouter><Login/></BrowserRouter> 
  );
  
  expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('render Home correctly', () => {
    const wrapper = shallow(
      <BrowserRouter><Home/></BrowserRouter> 
  );
  
  expect(toJson(wrapper)).toMatchSnapshot();
  });