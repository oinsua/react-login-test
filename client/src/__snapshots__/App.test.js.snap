// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Home correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Login />
</Router>
`;

exports[`render Login correctly 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Login />
</Router>
`;
